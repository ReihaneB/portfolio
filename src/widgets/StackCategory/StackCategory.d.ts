import { StaticImport } from 'next/dist/shared/lib/get-img-props';

interface tags {
  id: number;
  label: string;
  icon: StaticImport | null;
}

export interface StackCategoryProps {
  id?: number;
  name: string;
  tags: tags[];
}
