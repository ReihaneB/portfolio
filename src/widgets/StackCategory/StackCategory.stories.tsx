import type { Meta, StoryObj } from '@storybook/react';

import IconArrow from '@/styles/icons/arrows/arrow.svg';

import StackCategory from './StackCategory';

const meta: Meta<typeof StackCategory> = {
  title: 'Widgets/StackCategory',
  component: StackCategory,
};

export default meta;
type Story = StoryObj<typeof StackCategory>;

const defaultArgs = {
  stackCategory:
    {
      id: 1,
      name: 'Librairie',
      tags: [
        {
          id: 1,
          label: 'React',
          icon: IconArrow,
        },
        {
          id: 2,
          label: 'Redux',
          icon: IconArrow,
        },
        {
          id: 3,
          label: 'Jest',
          icon: IconArrow,
        },
        {
          id: 4,
          label: 'Next.js',
          icon: IconArrow,
        },
      ],
    },
};

export const Playground: Story = {
  args: {
    ...defaultArgs,
  },
};

export const WithLongName: Story = {
  args: {
    ...defaultArgs,
    stackCategory: {
      ...defaultArgs.stackCategory,
      name: 'Duis adipisicing ipsum in exercitation ut tempor voluptate eiusmod adipisicing.',
    },
  },
};

export const WithLongTags: Story = {
  args: {
    ...defaultArgs,
    stackCategory: {
      ...defaultArgs.stackCategory,
      tags: [
        {
          id: 1,
          label: 'React',
          icon: IconArrow,
        },
        {
          id: 2,
          label: 'Redux',
          icon: IconArrow,
        },
        {
          id: 3,
          label: 'Jest',
          icon: IconArrow,
        },
        {
          id: 4,
          icon: IconArrow,
          label: 'Enzyme',
        },
        {
          id: 5,
          label: 'Webpack',
          icon: IconArrow,
        },
        {
          id: 6,
          label: 'Babel',
          icon: IconArrow,
        },
        {
          id: 7,
          label: 'ESLint',
          icon: IconArrow,
        },
        {
          id: 8,
          label: 'Prettier',
          icon: IconArrow,
        },
        {
          id: 9,
          label: 'TypeScript',
          icon: IconArrow,
        },
        {
          id: 10,
          label: 'Jest',
          icon: IconArrow,
        },
        {
          id: 11,
          label: 'Enzyme',
          icon: IconArrow,
        },
      ],
    },
  },
};
