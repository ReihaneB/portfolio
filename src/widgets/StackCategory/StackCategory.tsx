import Tag from '@/components/Tag/Tag';

import type { StackCategoryProps } from './StackCategory.d';
import styles from './StackCategory.module.css';

function StackCategory({
  stackCategory,
}: StackCategoryProps) {
  return (
    <div>
      <h3 className={styles.name}>
        {stackCategory.name}
      </h3>
      <ul className={styles.tags}>
        {stackCategory.tags.map(tag => (
          <li key={tag.id}>
            <Tag Icon={tag.icon}>
              {tag.label}
            </Tag>
          </li>
        ))}
      </ul>
    </div>
  );
}

export default StackCategory;
