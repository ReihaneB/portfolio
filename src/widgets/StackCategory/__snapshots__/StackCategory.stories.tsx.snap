// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Widgets/StackCategory Playground smoke-test 1`] = `
<main class="-storybook-preview-module__root--fQckk inter-normal">
  <h3 class="src-widgets-StackCategory-StackCategory-module__name--XQSqW">
    Librairie
  </h3>
  <ul class="src-widgets-StackCategory-StackCategory-module__tags--XFK0O">
    <li>
      <div class="src-components-Tag-Tag-module__rootStyle--Cd2YJ"
           data-testid="tag-component"
           tabindex="-1"
      >
        <img alt="Icône de React"
             loading="lazy"
             width="12"
             height="12"
             decoding="async"
             data-nimg="1"
             class="src-components-Tag-Tag-module__icon--kRFOB"
             srcset="/static/media/arrow.55ce8cbb.svg?w=16&amp;q=75 1x, /static/media/arrow.55ce8cbb.svg?w=32&amp;q=75 2x"
             src="/static/media/arrow.55ce8cbb.svg?w=32&amp;q=75"
             style="color: transparent;"
        >
        React
      </div>
    </li>
    <li>
      <div class="src-components-Tag-Tag-module__rootStyle--Cd2YJ"
           data-testid="tag-component"
           tabindex="-1"
      >
        <img alt="Icône de Redux"
             loading="lazy"
             width="12"
             height="12"
             decoding="async"
             data-nimg="1"
             class="src-components-Tag-Tag-module__icon--kRFOB"
             srcset="/static/media/arrow.55ce8cbb.svg?w=16&amp;q=75 1x, /static/media/arrow.55ce8cbb.svg?w=32&amp;q=75 2x"
             src="/static/media/arrow.55ce8cbb.svg?w=32&amp;q=75"
             style="color: transparent;"
        >
        Redux
      </div>
    </li>
    <li>
      <div class="src-components-Tag-Tag-module__rootStyle--Cd2YJ"
           data-testid="tag-component"
           tabindex="-1"
      >
        <img alt="Icône de Jest"
             loading="lazy"
             width="12"
             height="12"
             decoding="async"
             data-nimg="1"
             class="src-components-Tag-Tag-module__icon--kRFOB"
             srcset="/static/media/arrow.55ce8cbb.svg?w=16&amp;q=75 1x, /static/media/arrow.55ce8cbb.svg?w=32&amp;q=75 2x"
             src="/static/media/arrow.55ce8cbb.svg?w=32&amp;q=75"
             style="color: transparent;"
        >
        Jest
      </div>
    </li>
    <li>
      <div class="src-components-Tag-Tag-module__rootStyle--Cd2YJ"
           data-testid="tag-component"
           tabindex="-1"
      >
        <img alt="Icône de Next.js"
             loading="lazy"
             width="12"
             height="12"
             decoding="async"
             data-nimg="1"
             class="src-components-Tag-Tag-module__icon--kRFOB"
             srcset="/static/media/arrow.55ce8cbb.svg?w=16&amp;q=75 1x, /static/media/arrow.55ce8cbb.svg?w=32&amp;q=75 2x"
             src="/static/media/arrow.55ce8cbb.svg?w=32&amp;q=75"
             style="color: transparent;"
        >
        Next.js
      </div>
    </li>
  </ul>
</main>
`;

exports[`Widgets/StackCategory WithLongName smoke-test 1`] = `
<main class="-storybook-preview-module__root--fQckk inter-normal">
  <h3 class="src-widgets-StackCategory-StackCategory-module__name--XQSqW">
    Duis adipisicing ipsum in exercitation ut tempor voluptate eiusmod adipisicing.
  </h3>
  <ul class="src-widgets-StackCategory-StackCategory-module__tags--XFK0O">
    <li>
      <div class="src-components-Tag-Tag-module__rootStyle--Cd2YJ"
           data-testid="tag-component"
           tabindex="-1"
      >
        <img alt="Icône de React"
             loading="lazy"
             width="12"
             height="12"
             decoding="async"
             data-nimg="1"
             class="src-components-Tag-Tag-module__icon--kRFOB"
             srcset="/static/media/arrow.55ce8cbb.svg?w=16&amp;q=75 1x, /static/media/arrow.55ce8cbb.svg?w=32&amp;q=75 2x"
             src="/static/media/arrow.55ce8cbb.svg?w=32&amp;q=75"
             style="color: transparent;"
        >
        React
      </div>
    </li>
    <li>
      <div class="src-components-Tag-Tag-module__rootStyle--Cd2YJ"
           data-testid="tag-component"
           tabindex="-1"
      >
        <img alt="Icône de Redux"
             loading="lazy"
             width="12"
             height="12"
             decoding="async"
             data-nimg="1"
             class="src-components-Tag-Tag-module__icon--kRFOB"
             srcset="/static/media/arrow.55ce8cbb.svg?w=16&amp;q=75 1x, /static/media/arrow.55ce8cbb.svg?w=32&amp;q=75 2x"
             src="/static/media/arrow.55ce8cbb.svg?w=32&amp;q=75"
             style="color: transparent;"
        >
        Redux
      </div>
    </li>
    <li>
      <div class="src-components-Tag-Tag-module__rootStyle--Cd2YJ"
           data-testid="tag-component"
           tabindex="-1"
      >
        <img alt="Icône de Jest"
             loading="lazy"
             width="12"
             height="12"
             decoding="async"
             data-nimg="1"
             class="src-components-Tag-Tag-module__icon--kRFOB"
             srcset="/static/media/arrow.55ce8cbb.svg?w=16&amp;q=75 1x, /static/media/arrow.55ce8cbb.svg?w=32&amp;q=75 2x"
             src="/static/media/arrow.55ce8cbb.svg?w=32&amp;q=75"
             style="color: transparent;"
        >
        Jest
      </div>
    </li>
    <li>
      <div class="src-components-Tag-Tag-module__rootStyle--Cd2YJ"
           data-testid="tag-component"
           tabindex="-1"
      >
        <img alt="Icône de Next.js"
             loading="lazy"
             width="12"
             height="12"
             decoding="async"
             data-nimg="1"
             class="src-components-Tag-Tag-module__icon--kRFOB"
             srcset="/static/media/arrow.55ce8cbb.svg?w=16&amp;q=75 1x, /static/media/arrow.55ce8cbb.svg?w=32&amp;q=75 2x"
             src="/static/media/arrow.55ce8cbb.svg?w=32&amp;q=75"
             style="color: transparent;"
        >
        Next.js
      </div>
    </li>
  </ul>
</main>
`;

exports[`Widgets/StackCategory WithLongTags smoke-test 1`] = `
<main class="-storybook-preview-module__root--fQckk inter-normal">
  <h3 class="src-widgets-StackCategory-StackCategory-module__name--XQSqW">
    Librairie
  </h3>
  <ul class="src-widgets-StackCategory-StackCategory-module__tags--XFK0O">
    <li>
      <div class="src-components-Tag-Tag-module__rootStyle--Cd2YJ"
           data-testid="tag-component"
           tabindex="-1"
      >
        <img alt="Icône de React"
             loading="lazy"
             width="12"
             height="12"
             decoding="async"
             data-nimg="1"
             class="src-components-Tag-Tag-module__icon--kRFOB"
             srcset="/static/media/arrow.55ce8cbb.svg?w=16&amp;q=75 1x, /static/media/arrow.55ce8cbb.svg?w=32&amp;q=75 2x"
             src="/static/media/arrow.55ce8cbb.svg?w=32&amp;q=75"
             style="color: transparent;"
        >
        React
      </div>
    </li>
    <li>
      <div class="src-components-Tag-Tag-module__rootStyle--Cd2YJ"
           data-testid="tag-component"
           tabindex="-1"
      >
        <img alt="Icône de Redux"
             loading="lazy"
             width="12"
             height="12"
             decoding="async"
             data-nimg="1"
             class="src-components-Tag-Tag-module__icon--kRFOB"
             srcset="/static/media/arrow.55ce8cbb.svg?w=16&amp;q=75 1x, /static/media/arrow.55ce8cbb.svg?w=32&amp;q=75 2x"
             src="/static/media/arrow.55ce8cbb.svg?w=32&amp;q=75"
             style="color: transparent;"
        >
        Redux
      </div>
    </li>
    <li>
      <div class="src-components-Tag-Tag-module__rootStyle--Cd2YJ"
           data-testid="tag-component"
           tabindex="-1"
      >
        <img alt="Icône de Jest"
             loading="lazy"
             width="12"
             height="12"
             decoding="async"
             data-nimg="1"
             class="src-components-Tag-Tag-module__icon--kRFOB"
             srcset="/static/media/arrow.55ce8cbb.svg?w=16&amp;q=75 1x, /static/media/arrow.55ce8cbb.svg?w=32&amp;q=75 2x"
             src="/static/media/arrow.55ce8cbb.svg?w=32&amp;q=75"
             style="color: transparent;"
        >
        Jest
      </div>
    </li>
    <li>
      <div class="src-components-Tag-Tag-module__rootStyle--Cd2YJ"
           data-testid="tag-component"
           tabindex="-1"
      >
        <img alt="Icône de Enzyme"
             loading="lazy"
             width="12"
             height="12"
             decoding="async"
             data-nimg="1"
             class="src-components-Tag-Tag-module__icon--kRFOB"
             srcset="/static/media/arrow.55ce8cbb.svg?w=16&amp;q=75 1x, /static/media/arrow.55ce8cbb.svg?w=32&amp;q=75 2x"
             src="/static/media/arrow.55ce8cbb.svg?w=32&amp;q=75"
             style="color: transparent;"
        >
        Enzyme
      </div>
    </li>
    <li>
      <div class="src-components-Tag-Tag-module__rootStyle--Cd2YJ"
           data-testid="tag-component"
           tabindex="-1"
      >
        <img alt="Icône de Webpack"
             loading="lazy"
             width="12"
             height="12"
             decoding="async"
             data-nimg="1"
             class="src-components-Tag-Tag-module__icon--kRFOB"
             srcset="/static/media/arrow.55ce8cbb.svg?w=16&amp;q=75 1x, /static/media/arrow.55ce8cbb.svg?w=32&amp;q=75 2x"
             src="/static/media/arrow.55ce8cbb.svg?w=32&amp;q=75"
             style="color: transparent;"
        >
        Webpack
      </div>
    </li>
    <li>
      <div class="src-components-Tag-Tag-module__rootStyle--Cd2YJ"
           data-testid="tag-component"
           tabindex="-1"
      >
        <img alt="Icône de Babel"
             loading="lazy"
             width="12"
             height="12"
             decoding="async"
             data-nimg="1"
             class="src-components-Tag-Tag-module__icon--kRFOB"
             srcset="/static/media/arrow.55ce8cbb.svg?w=16&amp;q=75 1x, /static/media/arrow.55ce8cbb.svg?w=32&amp;q=75 2x"
             src="/static/media/arrow.55ce8cbb.svg?w=32&amp;q=75"
             style="color: transparent;"
        >
        Babel
      </div>
    </li>
    <li>
      <div class="src-components-Tag-Tag-module__rootStyle--Cd2YJ"
           data-testid="tag-component"
           tabindex="-1"
      >
        <img alt="Icône de ESLint"
             loading="lazy"
             width="12"
             height="12"
             decoding="async"
             data-nimg="1"
             class="src-components-Tag-Tag-module__icon--kRFOB"
             srcset="/static/media/arrow.55ce8cbb.svg?w=16&amp;q=75 1x, /static/media/arrow.55ce8cbb.svg?w=32&amp;q=75 2x"
             src="/static/media/arrow.55ce8cbb.svg?w=32&amp;q=75"
             style="color: transparent;"
        >
        ESLint
      </div>
    </li>
    <li>
      <div class="src-components-Tag-Tag-module__rootStyle--Cd2YJ"
           data-testid="tag-component"
           tabindex="-1"
      >
        <img alt="Icône de Prettier"
             loading="lazy"
             width="12"
             height="12"
             decoding="async"
             data-nimg="1"
             class="src-components-Tag-Tag-module__icon--kRFOB"
             srcset="/static/media/arrow.55ce8cbb.svg?w=16&amp;q=75 1x, /static/media/arrow.55ce8cbb.svg?w=32&amp;q=75 2x"
             src="/static/media/arrow.55ce8cbb.svg?w=32&amp;q=75"
             style="color: transparent;"
        >
        Prettier
      </div>
    </li>
    <li>
      <div class="src-components-Tag-Tag-module__rootStyle--Cd2YJ"
           data-testid="tag-component"
           tabindex="-1"
      >
        <img alt="Icône de TypeScript"
             loading="lazy"
             width="12"
             height="12"
             decoding="async"
             data-nimg="1"
             class="src-components-Tag-Tag-module__icon--kRFOB"
             srcset="/static/media/arrow.55ce8cbb.svg?w=16&amp;q=75 1x, /static/media/arrow.55ce8cbb.svg?w=32&amp;q=75 2x"
             src="/static/media/arrow.55ce8cbb.svg?w=32&amp;q=75"
             style="color: transparent;"
        >
        TypeScript
      </div>
    </li>
    <li>
      <div class="src-components-Tag-Tag-module__rootStyle--Cd2YJ"
           data-testid="tag-component"
           tabindex="-1"
      >
        <img alt="Icône de Jest"
             loading="lazy"
             width="12"
             height="12"
             decoding="async"
             data-nimg="1"
             class="src-components-Tag-Tag-module__icon--kRFOB"
             srcset="/static/media/arrow.55ce8cbb.svg?w=16&amp;q=75 1x, /static/media/arrow.55ce8cbb.svg?w=32&amp;q=75 2x"
             src="/static/media/arrow.55ce8cbb.svg?w=32&amp;q=75"
             style="color: transparent;"
        >
        Jest
      </div>
    </li>
    <li>
      <div class="src-components-Tag-Tag-module__rootStyle--Cd2YJ"
           data-testid="tag-component"
           tabindex="-1"
      >
        <img alt="Icône de Enzyme"
             loading="lazy"
             width="12"
             height="12"
             decoding="async"
             data-nimg="1"
             class="src-components-Tag-Tag-module__icon--kRFOB"
             srcset="/static/media/arrow.55ce8cbb.svg?w=16&amp;q=75 1x, /static/media/arrow.55ce8cbb.svg?w=32&amp;q=75 2x"
             src="/static/media/arrow.55ce8cbb.svg?w=32&amp;q=75"
             style="color: transparent;"
        >
        Enzyme
      </div>
    </li>
  </ul>
</main>
`;
